rsocket-core/src/main/java/io/rsocket/RSocketFactory.java

```diff
@@ -257,15 +286,22 @@ public ClientRSocketFactory setupPayload(Payload payload) {
         private Consumer<Throwable> errorConsumer = Throwable::printStackTrace;
         private int mtu = 0;
 
-        private ServerRSocketFactory() {
-        }
+        private @Nullable RSocketStats stats;
 
         @Override
         public Transport<io.rsocket.transport.ServerTransport, Closeable> acceptor(Supplier<SocketAcceptor> acceptor) {
             this.acceptor = acceptor;
             return new ServerTransport();
         }
 
+        public ServerRSocketFactory stats(RSocketStats stats) {
+            this.stats = stats;
+            if (stats != null) {
+                stats.socketCreated();
```

Comment:
This seems like its per server, which can have at any time 0..n connections.  Should this be deferred to ServerStart::start where it receives a DuplexConnection?